x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -1181,6 +1186,9 @@ public int populateAndRead(
             final RangeAvailableHandler reader,
             final RangeMissingHandler writer
         ) throws Exception {
+            if (closed.get()) {
```

Comment:
Would it be more comfortable to expose `sharedBytes.fileChannel.isOpen()` and check if it's false here? That way, we avoid adding the additional `closed` flag.